### PR TITLE
[Ide] Fix various leaks causing retention of document model and autotest objects

### PR DESCRIPTION
**Pull request by Therzok**
Fixes VSTS #964955 - ObjectProperties are leaked due in AutoTest service
Fixes VSTS #965064 - GtkShellDocumentView is not properly disposed

### CODE REVIEW

**Comment by mrward** [comment]
Re-running the build. Looks like an intermittent NuGet error.

**Comment by Therzok** [comment]
Yeah, the fix isn't complete. I'm still trying to track down why the new UITest infrastructure is not properly clearing references, whilst the old one does.

**Comment by Therzok** [comment]
Should be ready now.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Mac/MDLinkMenuItem.cs [hunk 1]
```diff
@@ -40,14 +40,13 @@ public MDLinkMenuItem (LinkCommandEntry lce)
...
 		}
 
-		[Export (MDMenuItem.ActionSelName)]
-		public void Run (NSObject dummy)
+		static void MDLinkMenuItem_Activated (object sender, System.EventArgs e)
 		{
-			MonoDevelop.Ide.IdeServices.DesktopService.ShowUrl (lce.Url);
+			if (sender is MDLinkMenuItem linkMenuItem)
+			MonoDevelop.Ide.IdeServices.DesktopService.ShowUrl (linkMenuItem.lce.Url);
 		}
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Mac/MDLinkMenuItem.cs]: The indenting is a bit odd here.

**Comment by Therzok** [comment]
We might want this in 8.3 too, for the leak fix.

**Comment by sevoku** [comment]
@monojenkins backport release-8.3

**Pull request merged**